library/alloc/src/alloc.rs

```diff
@@ -172,6 +177,7 @@ pub unsafe fn alloc_zeroed(layout: Layout) -> *mut u8 {
 
 #[cfg(not(test))]
 impl Global {
+    #[requires(layout.size() == 0 || layout.align() != 0)]
```

Comment:
I thought we were not generating `requires` for safe functions.

Reply:
Thank you, I have now removed this and added an `ensures` clause for this one instead.